conda_forge_tick/git_utils.py

```diff
@@ -225,63 +254,114 @@ def _run_git_command(
-        If all_ is set with empty pathspec, all files in the entire working tree are updated.
-        :raises ValueError: If pathspec is empty and all_ is False.
-        :raises GitCliError: If the git command fails.
+        """Add files to the git index with `git add`.
+
+        Parameters
+        ----------
+        git_dir : str
+            The directory of the git repository.
+        pathspec : str
+            The files to add.
+        all_ : bool, optional
+            If True, not only add the files in pathspec, but also where the index
+            already has an entry. If all_ is set with empty pathspec, all files
+            in the entire working tree are updated.
+
+        Raises
+        ------
+        ValueError
+            If pathspec is empty and all_ is False.
+        GitCliError
+            If the git command fails.
         """
         if not pathspec and not all_:
             raise ValueError("Either pathspec or all_ must be set.")
 
         all_arg = ["--all"] if all_ else []
 
-        self._run_git_command(["add", *all_arg, *pathspec], git_dir)
+        try:
+            self._run_git_command(["add", *all_arg, *pathspec], git_dir)
+        except GitCliError as e:
+            raise GitCliError("Adding files to git failed.") from e
 
     @lock_git_operation()
     def commit(
         self, git_dir: Path, message: str, all_: bool = False, allow_empty: bool = False
     ):
-        """
-        Commit changes to the git repository with `git commit`.
-        :param git_dir: The directory of the git repository.
-        :param message: The commit message.
-        :param allow_empty: If True, allow an empty commit.
-        :param all_: Automatically stage files that have been modified and deleted, but new files are not affected.
-        :raises GitCliError: If the git command fails.
+        """Commit changes to the git repository with `git commit`.
+
+        Parameters
+        ----------
+        git_dir : Path
+            The directory of the git repository.
+        message : str
+            The commit message.
+        allow_empty : bool, optional
+            If True, allow an empty commit.
+        all_ : bool, optional
+            Automatically stage files that have been modified and deleted, but new
+            files are not affected.
+
+        Raises
+        ------
+        GitCliError
+            If the git command fails.
         """
         all_arg = ["-a"] if all_ else []
         allow_empty_arg = ["--allow-empty"] if allow_empty else []
 
-        self._run_git_command(
-            ["commit", *all_arg, *allow_empty_arg, "-m", message], git_dir
-        )
+        try:
+            self._run_git_command(
+                ["commit", *all_arg, *allow_empty_arg, "-m", message], git_dir
+            )
+        except GitCliError as e:
+            raise GitCliError("Could not commit.") from e
```

Comment:
Same here.

Reply:
This change is required because of the new `DOC501` and `DOC502` ruff rules. It doesn't change any behaviour, just makes existing errors more explicit.